com.unity.netcode.gameobjects/Tests/Runtime/OwnerPermissionTests.cs

```diff
@@ -0,0 +1,221 @@
+
+            MyNetworkListServer = new NetworkList<int>(writePerm: NetworkVariableWritePermission.Server);
+            MyNetworkListServer.OnListChanged += ListServerChanged;
+        }
+
+        public void OwnerChanged(int before, int after)
+        {
+        }
+
+        public void ServerChanged(int before, int after)
+        {
+        }
+
+        public void ListOwnerChanged(NetworkListEvent<int> listEvent)
+        {
+        }
+
+        public void ListServerChanged(NetworkListEvent<int> listEvent)
+        {
+        }
+    }
+
+    public class OwnerPermissionHideTests : NetcodeIntegrationTest
+    {
+        protected override int NumberOfClients => 2;
+
+        private GameObject m_PrefabToSpawn;
+
+        protected override void OnCreatePlayerPrefab()
+        {
+            var networkTransform = m_PlayerPrefab.AddComponent<OwnerPermissionTestComponent>();
+        }
+
+        protected override void OnServerAndClientsCreated()
+        {
+            m_PrefabToSpawn = CreateNetworkObjectPrefab("OwnerPermissionObject");
+            m_PrefabToSpawn.AddComponent<OwnerPermissionObject>();
+        }
+
+        [UnityTest]
+        public IEnumerator OwnerPermissionTest()
+        {
+            // create 3 objects
+            for (var objectIndex = 0; objectIndex < 3; objectIndex++)
+            {
+                OwnerPermissionObject.CurrentlySpawning = objectIndex;
+
+                NetworkManager ownerManager = m_ServerNetworkManager;
+                if (objectIndex != 0)
+                {
+                    ownerManager = m_ClientNetworkManagers[objectIndex - 1];
+                }
+                SpawnObject(m_PrefabToSpawn, ownerManager);
+
+                // wait for each object to spawn on each client
+                for (var clientIndex = 0; clientIndex < 3; clientIndex++)
+                {
+                    while (OwnerPermissionObject.Objects[objectIndex, clientIndex] == null)
+                    {
+                        yield return new WaitForSeconds(0.0f);
+                    }
+                }
+            }
+
+            var nextValueToWrite = 1;
+            var serverIndex = 0;
+
+            for (var objectIndex = 0; objectIndex < 3; objectIndex++)
+            {
+                for (var clientWriting = 0; clientWriting < 3; clientWriting++)
+                {
+                    // ==== Server-writable NetworkVariable ====
+                    var gotException = false;
+                    Debug.Log($"Writing to server-write variable on object {objectIndex} on client {clientWriting}");
+
+                    try
+                    {
+                        nextValueToWrite++;
+                        OwnerPermissionObject.Objects[objectIndex, clientWriting].MyNetworkVariableServer.Value = nextValueToWrite;
+                    }
+                    catch (Exception)
+                    {
+                        gotException = true;
```

Comment:
Out of curiosity, why not do this and provide the exception information?
```
            catch(Exception ex)
            {
                Assert.Fail(ex.Message);
            }
```

Reply:
Same for the remaining try-catch blocks below.

Reply:
Because a fair share of exception are expected in this test. See below lines:

```
Debug.Assert(gotException == (clientWriting != objectIndex));
``` 

I guess, though I could save the exception, and log it, if it is unexpected. But for the case where an exception is expected but was not received, it won't help.

Reply:
Right, was just pointing out that you can achieve the same result with catching exceptions and include the exception message without needing to use a bool to determine if you threw an exception or not.  
More of a style thing... so not anything I would block a PR for.